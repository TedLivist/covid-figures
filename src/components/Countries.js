import React, { useEffect } from 'react';
import { useDispatch } from 'react-redux';
import { getCountries } from '../redux/countries/countries';
import NewComp from './NewComp';

const Countries = () => {
  const dispatch = useDispatch()
  useEffect(() => {
    dispatch(getCountries())
  }, [dispatch])
  return (
    <div>
      <p>Haha Countries</p>
      <NewComp />
    </div>
  );
}
 
export default Countries;