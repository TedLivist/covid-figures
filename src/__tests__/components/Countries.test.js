import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Countries from '../../components/Countries';
import store from '../../redux/configureStore';

test('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Countries />
        </BrowserRouter>
      </Provider>
      )
    .toJSON()
  expect(tree).toMatchSnapshot()
})