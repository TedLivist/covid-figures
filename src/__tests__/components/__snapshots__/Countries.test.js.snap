// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="d-flex fixed-top header px-2 py-2"
  >
    <div
      className="d-flex"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        <box-icon
          color="white"
          name="chevron-left"
          type="solid"
        />
      </a>
      <div
        className="text-white"
      >
        2021
      </div>
    </div>
    <div
      className="text-white"
    >
      countries cases
    </div>
    <div
      className="d-flex"
    >
      <div
        className="me-4"
      >
        <box-icon
          color="white"
          name="microphone"
        />
      </div>
      <div>
        <box-icon
          color="white"
          name="cog"
        />
      </div>
    </div>
  </div>
  <div
    className="jumbo d-flex"
  >
    <div
      className="icon-div w-50"
    >
      <box-icon
        color="#7B2C4B"
        name="world"
        size="8rem"
      />
    </div>
    <div
      className="w-50 jumbo-title"
    >
      <div>
        Countries
      </div>
      <div />
    </div>
  </div>
  <div
    className="topic-input d-flex"
  >
    <div
      className="topic-div"
    >
      <div>
        CASES BY COUNTRY
      </div>
    </div>
    <div
      className="search-input pe-3"
    >
      <input
        onChange={[Function]}
        placeholder="Search movie"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="countries-container"
  />
</div>
`;
